### PR TITLE
feat: add database postgres

### PR DESCRIPTION
**Pull request by patriciapedrosaa**
## Context 

This PR provides storage using postgres with the pgxpool lib and automates the execution of migrations. Parameterizing the use of context in use cases, repositories and handlers. Finally, it adjusts the unit tests changed with these changes. 

## Changes

- [x] Add migrations
- [x] Add postgres database to accounts
- [x] Add postgres database to transfers
- [x] Add postgres database to autehntication
- [x] Add context to account use cases, database and handlers
- [x] Add contect to transfers use cases, database and handlers
- [x] Add context to autehtnticatio use cases, database and handlers

### CODE REVIEW

**Review by pedroyremolo — COMMENTED** [review]
just some small comments to entertain yourself

File: app/domain/account/usecase/get_balance_test.go [hunk 1]
```diff
@@ -1,43 +1,74 @@
...
-	accountUseCase := NewAccountUseCase(&memoryStorage, zerolog.Logger{})
-	fakeAccount1 := account.CreateAccountInput{
-		Name:   "Sirius Black",
-		CPF:    "12345678911",
-		Secret: "foobar",
-	}
-	account1, _ := accountUseCase.Create(fakeAccount1)
+const (
+	fakeID      = "deef3e68-a8ed-4320-b09c-037eab340125"
+	wrongFakeID = "deef3e68-a8ed-4320-b09c-037eab340125"
```
[the remark this answers is not in the document]
Comment (patriciapedrosaa) [reply app/domain/account/usecase/get_balance_test.go:14]: I forgot to change!

File: app/domain/account/usecase/get_test.go [hunk 2]
```diff
@@ -48,10 +48,20 @@ func TestGetAccounts(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:          "should return an error because repository returned an unexpected err",
+			repositoryErr: unexpectedRepositoryErr,
+			wantErr:       nil,
```
[the remark this answers is not in the document]
Comment (patriciapedrosaa) [reply app/domain/account/usecase/get_test.go:54]: adjusted!

File: app/domain/authentication/usecase/validates_token_test.go [hunk 3]
```diff
@@ -134,3 +134,51 @@ func generateFakeToken() string {
...
+//t.Run("Should return error because token is not in database", func(t *testing.T) {
+//	fakeToken := generateFakeToken()
+//	wantErr := ErrTokenNotFound
+//
+//	got, err := authenticationUseCase.ValidatesToken(fakeToken)
+//
+//	assert.Empty(t, got)
+//	assert.Error(t, err)
+//	assert.Equal(t, wantErr, err)
+//})
```
[the remark this answers is not in the document]
Comment (patriciapedrosaa) [reply app/domain/authentication/usecase/validates_token_test.go:184]: I will delete!

File: app/domain/account/usecase/get_test.go [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
+		CPF:     "12345678910",
+		Secret:  "foobar",
+		Balance: 100,
 	}
-	fakeAccount2 := account.CreateAccountInput{
-		Name:   "Ginevra Weasley",
-		CPF:    "12345678911",
-		Secret: "foobar",
+	fakeAccount2 = entities.Account{
+		Name:    "Ginevra Weasley",
```
Comment (erikacarvalho) [inline app/domain/account/usecase/get_test.go:20]: love the account owners

File: app/domain/transfer/usecase/get_by_account_id.go [hunk 5]
```diff
@@ -1,16 +1,28 @@
...
 	log := t.logger.With().
 		Str("account_ID", accountID).
 		Logger()
 
 	log.Info().Msg("getting transfers for account id")
-	transfers, err := t.transferRepository.GetTransfersByAccountID(accountID)
+
+	_, err := uuid.Parse(accountID)
+	if err != nil {
+		log.Error().Err(err).Msg(" error occurred when was trying get transfer for account id.")
```
Comment (pedroyremolo) [inline app/domain/transfer/usecase/get_by_account_id.go:21]: same as before

File: app/gateways/http/transfer/get.go [hunk 6]
```diff
@@ -15,14 +18,23 @@ type GetTransferResponse struct {
...
+			h.logger.Err(err).
+				Int("status_code", http.StatusBadRequest).
+				Msg("error occurred when try to get transfers")
+			http_server.ResponseError(w, http.StatusBadRequest, ErrInvalidId)
+		default:
+			h.logger.Err(err).
+				Int("status_code", http.StatusBadRequest).
+				Msg("error occurred when try to get transfers")
+			http_server.ResponseError(w, http.StatusBadRequest, ErrUnexpected)
+		}
```
Comment (pedroyremolo) [inline app/gateways/http/transfer/get.go:37]: well, if you're throwing an unexpected error, why would you say it's the client's fault?

File: cmd/main.go [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+func main() {
+	//Log
+	logger := zerolog.New(os.Stdout).With().Timestamp().Logger()
+
+	//Memory
+
+	//accountStorage := make(map[string]memory.Account)
+	//transferStorage := make(map[string][]memory.Transfer)
+	//authenticationStorage := make(map[string]memory.Token)
+	//memoryStorage := memory.NewMemoryStorage(accountStorage, transferStorage, authenticationStorage)
```
Comment (pedroyremolo) [inline cmd/main.go:41]: remove unused code to avoid confusion

**Review by pedroyremolo — COMMENTED** [review]
another round 🥊

**Review by pedroyremolo — APPROVED** [review]

**Pull request merged**